enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,114 @@
+# Node Autoreplacement
+
+## Summary
+
+Node Autoreplacement enhancement extends Scylla Operator with a functionality of automated replacement of nodes which lost their storage, which, at this point, renders them getting stuck on joining the cluster.
+
+This proposal suggests creating a module responsible for triggering a node replacement in such circumstances.
+As a side effect, introducing such functionality would allow for simplifying the existing `OrphanedPVController` by
+reducing its task to evicting orphaned PVs.
+
+## Motivation
+
+Currently, when a node looses its storage, it gets stuck on joining a cluster, since Scylla produces a following runtime error:
+```
+Startup failed: std::runtime_error (A node with address <ip> already exists, cancelling join. Use replace_address if you want to replace this node.)
+```
+
+Successfully restarting the node requires a manual intervention of a human operator by annotating the
+service corresponding to the affected node with a `scylla/replace` label. The main goal of this proposal is to improve
+Scylla Operator's stability in running Scylla independently of its administrators' interventions.
+
+### Goals
+
+- Successfully discovering and autoreplacing nodes which lost their storage without requiring user intervention.
+
+### Non-Goals
+
+- Removing orphaned PVs. This is left for `OrphanedPVController`.
+
+## Proposal
+
+Create an InitContainer with a dedicated Scylla Operator command responsible for checking whether a given node is
+1. **not** first in the cluster
```

Comment:
first in the sense of having index 0, or first bootstrapping node?

Reply:
first bootstrapping node, see https://github.com/scylladb/scylla/blob/b503ec36c22856cf9e1d02ff0055767001be6f66/service/storage_service.cc#L240

Reply:
the answer was meant to be included in the document so it's not ambiguous